### ISSUE TITLE
Use of instanceof during static initialization

### ISSUE THREAD

**Issue by mckuhei** [leaningtech/cheerpj-applet-runner#61]
![image](https://user-images.githubusercontent.com/31839741/229552455-1329165b-7cae-4ae3-9ed6-3f1f5e095328.png)
![image](https://user-images.githubusercontent.com/31839741/229552505-6815a3f1-9f4d-46cd-8d32-cb60374416f8.png)
![image](https://user-images.githubusercontent.com/31839741/229552604-7b061b94-b52f-477a-bf30-c33d8e1f668e.png)
![image](https://user-images.githubusercontent.com/31839741/229552650-4d025eef-841e-4983-ada4-8cb9e3e21157.png)

**Comment by alexp-sssup** [comment]
Instanceof is certainly correctly implemented for this basic case. Please provide a fully reproducible *minimal* example.

**Comment by mckuhei** [comment]
BlockMushroom is extend BlockFlower, And BlockFlower is extend Block, But when use `Object obj = new BlockMushroom()` then `obj instanceof Block` return false;

**Comment by alexp-sssup** [comment]
Can you actually provide a single self contained .java file that demonstrate the problem?

Instanceof is a fundamental part of the Java language, and it is used extensively in the Java runtime. It is simply not possible that the problem is what you describe.

**Comment by mckuhei** [comment]
Sorry, I already made a demo, But it works fine. I don't why. But I removed instanceof block and direct cast to Block. It's fine. I will upload a jar file with sources sometime.

**Comment by mckuhei** [comment]
For temporary fix, Use obj.getClass()..isSubClass(), I will upload jar and sources sometime.

**Comment by mckuhei** [comment]
![image](https://user-images.githubusercontent.com/31839741/229652865-79a0ad1c-9d5f-4f82-b344-d3c5eadef90e.png)
![image](https://user-images.githubusercontent.com/31839741/229652888-cc5df92d-d021-466c-b5fd-86c7145ae923.png)
![image](https://user-images.githubusercontent.com/31839741/229653002-4d0ee9c0-6e21-4030-b465-a19ef68e62af.png)
![image](https://user-images.githubusercontent.com/31839741/229653048-1d50781d-4462-4d70-9a38-3c2acc245a41.png)
That's weird, it doesn't look like it's guaranteed to recur.

**Comment by mckuhei** [comment]
Jar an sources: https://www.mediafire.com/file/wwkp3s98ye1931t/map.zip/file
NOTE: Please notify me after downloaded. Use TestApplet or IsomPreviewApplet.

**Comment by alexp-sssup** [comment]
Please provide the HTML and the jar.js files you are using as well, if any

**Comment by mckuhei** [comment]
HTML:
```html
<html>
<head>
<title>Minecraft infdev map visualiser</title>
</head>
<body bgcolor="#000000" text="#906060">
<center>
<applet code="net.minecraft.isom.IsomPreviewApplet" archive="map.jar?v=6" width="1280" height="720">
</applet>
</center>
</body>
</html>
```
I use browser extension.

**Comment by mckuhei** [comment]
Okay, Thank you!

**Comment by alexp-sssup** [comment]
This bug is fixed in version `2024.10` of the CheerpJ Applet Runner extension. This new version is based on CheerpJ 3.0 architecture and brings many improvements to performance and robustness.

**Comment by mckuhei** [comment]
Thank you!
